src/Adapter/Product/Combination/CommandHandler/RemoveAllAssociatedCombinationSuppliersHandler.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\Combination\CommandHandler;
+
+use PrestaShop\PrestaShop\Adapter\Product\Combination\Repository\CombinationRepository;
+use PrestaShop\PrestaShop\Adapter\Product\Update\ProductSupplierUpdater;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Command\RemoveAllAssociatedCombinationSuppliersCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\CommandHandler\RemoveAllAssociatedCombinationSuppliersHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+
+/**
+ * Removes product suppliers associated to a certain combination using legacy object model
+ */
+final class RemoveAllAssociatedCombinationSuppliersHandler implements RemoveAllAssociatedCombinationSuppliersHandlerInterface
+{
+    /**
+     * @var CombinationRepository
+     */
+    private $combinationRepository;
+
+    /**
+     * @var ProductSupplierUpdater
+     */
+    private $productSupplierUpdater;
+
+    /**
+     * @param CombinationRepository $combinationRepository
+     * @param ProductSupplierUpdater $productSupplierUpdater
+     */
+    public function __construct(
+        CombinationRepository $combinationRepository,
+        ProductSupplierUpdater $productSupplierUpdater
+    ) {
+        $this->productSupplierUpdater = $productSupplierUpdater;
+        $this->combinationRepository = $combinationRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(RemoveAllAssociatedCombinationSuppliersCommand $command): void
+    {
+        $combination = $this->combinationRepository->get($command->getCombinationId());
```

Comment:
hmm In this case (combination handlers) the first thing that it does is tries to load the combination, so if there is none, it will throw the exception. (because we don't even provide ProductId in this command).
in normal handlers additional check should help tho.